shm/Map.hpp

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#include <algorithm>
+#include <iostream>
+#include <random>
+#include <vector>
+
+#include "Island.hpp"
+
+constexpr size_t defaultXsize = 10;
+constexpr size_t defaultYsize = 11;
+
+class Map
+{
+
+public:
+    Map() : mapXsize(defaultXsize), mapYsize(defaultYsize) { randomIslandGenerator(10); }
+    Map(int islandsAmount) { randomIslandGenerator(islandsAmount); }
+    Map(int sizeY, int sizeX, int islandsAmount) : mapXsize(sizeY), mapYsize(sizeX)
+    {
+        randomIslandGenerator(islandsAmount);
+    }
```

Comment:
You really need all of this constructor? Maybe try to simply it? Also try reuse other constructor.